Source/DafnyCore/ParserErrorDetail.cs

```diff
@@ -12,50 +12,361 @@ public class ParserErrorDetail {
 
   public static void init() {
 
-    ErrorDetail.Add(ErrorID.p_bad_const_initialize_op,
+    Add(ErrorID.p_duplicate_modifier,
```

Comment:
I think it's better to have the explanations be located where the error messages are also created, so the code remains local.

Reply:
We debated this too. At least for the batch that are parser error explanations, it seemed to make the parser code cluttered to insert explanations in the code for all the productions. And the other extreme, one location for all explanations seemed too non-local. So we compromised by having one file that is in close association with the code whose errors it explains.

Reply:
What do you mean by cluttered?

`Dafny.atg` is too large and centralised as is, and ideally we break it up into pieces, but that's something for the future. For now I think adding the extra information in `Dafny.atg` is OK, but I'm also OK with having a separate file for error information if it only relates to the parser.

Reply:
I agree with David. Because explanations can be heavy, it's better if they are in a separate file so that they don't distract a developer who thinks about the code.
It's the same situation as for the ProofObligationDescription, we move the wording in separate classes so that, even though we don't have the wording locally when creating the assertion, it's still easily accessible, and some elements can also be factored together.